### DISCUSSION TITLE
Faulting a workflow: FaultActivity

### DISCUSSION THREAD

**Discussion by terry-delph** [elsa-workflows/elsa-core#3032] [Q&A] [answered]
We have some use cases where we want to process the following business logic within a workflow.

1. Loop through a list, capture any of the iterations outcomes that were deemed an error in a variable
2. Loop completes - some may be in state of error, some may not
3. Check if any of the iterations had an outcome that was deemed an error
4. If any were deemed an error, Fault the workflow so that it ends

### Step 4 Questions:
**Is there a a built in activity to do this?**
I thought the _Finish_ activity may do this, but it does not look like it.

**Is the following activity implementation appropriate?**
```
[Activity(
        Category = "Workflow State",
        Description = "Fault the workflow",
        DisplayName = "Fault Workflow")]
    public class ThrowFault : Activity
    {
        [ActivityInput(
            Label = "Fault Message",
            Hint = "Message to fault the workflow with.",
            SupportedSyntaxes = new[] { SyntaxNames.JavaScript, SyntaxNames.Liquid })]
        public string FaultMessage { get; set; }
        public override async ValueTask<IActivityExecutionResult> ExecuteAsync(ActivityExecutionContext context)
        {
            context.JournalData.Add("FaultMessage", FaultMessage);
            return Fault(FaultMessage);
        }
    }
```
Is this possibly an activity that should be part of the core ELSA activities?

Kind regards

**Comment by sfmskywalker** [comment, marked as answer]
We recently [added a new set of activities](https://github.com/elsa-workflows/elsa-core/pull/2940) (to be released with 2.7, currently available from MyGet), one of them being the [Fault activity](https://github.com/elsa-workflows/elsa-core/blob/master/src/core/Elsa.Core/Activities/Primitives/Fault/Fault.cs) - which pretty much does exactly what your `ThrowFault` activity is doing :)
